.github/workflows/flags.yml

```diff
@@ -0,0 +1,165 @@
+          echo "CABAL_PKGS_DIR=~/.cabal/packages" >> $GITHUB_ENV
+
+      # All workflows which distinquishes cache on `cabal.project` needs this.
+      - name: Workaround shorten binary names
+        run: |
+          sed -i.bak -e 's/haskell-language-server/hls/g' \
+                     -e 's/haskell_language_server/hls/g' \
+                     haskell-language-server.cabal cabal.project
+          sed -i.bak -e 's/Paths_haskell_language_server/Paths_hls/g' \
+                     src/**/*.hs exe/*.hs
+
+      - name: Retrieving `cabal.project` Hackage timestamp
+        run: |
+          # Form: index-state: 2021-11-29T08:11:08Z
+          INDEX_STATE_ENTRY=$(grep index-state cabal.project)
+          # Form: 2021-11-29T08-11-08Z
+          INDEX_STATE1=$(echo "$INDEX_STATE_ENTRY" | cut -d' ' -f2 | tr ':' '-')
+          echo "INDEX_STATE=$INDEX_STATE1" >> $GITHUB_ENV
+
+      - name: Form the package list ('cabal.project.freeze')
+        continue-on-error: true
+        run: |
+          cabal v2-freeze && \
+          echo "" && \
+          echo 'Output:' && \
+          echo "" && \
+          cat 'cabal.project.freeze' && \
+          echo "" || \
+          echo 'WARNING: Could not produce the `freeze`.'
+
+      - name: Hackage sources cache
+        uses: actions/cache@v2
+        env:
+          cache-name: hackage-sources
+        with:
+          path: ${{ env.CABAL_PKGS_DIR }}
+          key:          ${{ env.cache-name }}-${{ env.INDEX_STATE }}
+          restore-keys: ${{ env.cache-name }}-
+
+      - name: Compiled deps cache
+        id: compiled-deps
+        uses: actions/cache@v2
+        env:
+          cache-name: compiled-deps
+        with:
+          path: ${{ steps.HaskEnvSetup.outputs.cabal-store }}
+          key:  ${{ env.cache-name }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.INDEX_STATE }}-${{ hashFiles('cabal.project.freeze') }}
+          restore-keys: |
+                ${{ env.cache-name }}-${{ runner.os }}-${{ matrix.ghc }}-${{ env.INDEX_STATE }}-
+                ${{ env.cache-name }}-${{ runner.os }}-${{ matrix.ghc }}-
+                ${{ env.cache-name }}-${{ runner.os }}-
+
+      - if: needs.pre_job.outputs.should_skip != 'true'
+        name: Build `hls-graph` with flags
+        run: cabal v2-build hls-graph --flags="pedantic embed-files stm-stats"
+
+      - if: needs.pre_job.outputs.should_skip != 'true'
+        name: Build `hie-compat` with flags
+        run: cabal v2-build hie-compat --flags="ghc-lib"
+
+      - if: needs.pre_job.outputs.should_skip != 'true'
+        name: Build `hls-plugin-api` with flags
+        run: cabal v2-build hls-plugin-api --flags="pedantic"
+
+      - if: needs.pre_job.outputs.should_skip != 'true'
+        name: Build `hls-test-utils` with flags
+        run: cabal v2-build hls-test-utils --flags="pedantic"
+
+      # repeating builds to workaround segfaults in windows and ghc-8.8.4
+      - if: needs.pre_job.outputs.should_skip_ghcide != 'true'
```

Comment:
we are only in ubuntu here, so not need for three builds

Reply:
Yes.

I still do this with deduplication in mind (doing deduplication locally).